core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -455,4 +459,16 @@ public static void parseCopyFields(Object propNode, FieldMapper.Builder builder)
         builder.copyTo(copyToBuilder.build());
     }
 
+    private static SimilarityProvider resolveSimilarity(Mapper.TypeParser.ParserContext parserContext, String name, String value) {
+        if (parserContext.indexVersionCreated().before(Version.V_3_0_0) &&
+            "default".equals(value) && parserContext.getSimilarity(value) == null) {
```

Comment:
in a previous comment I thought you had said that the parserContext.getSimilarity part was not relevant anymore?